### ISSUE TITLE
Grayscale jpg to pdf - wrong pdf file.

### ISSUE THREAD

**Issue by dymn** [ImageMagick/ImageMagick#4739]
### ImageMagick version

7.1.0.13

### Operating system

Linux

### Operating system, version and so on

Gentoo

### Description

When I try to convert one or more grayscale jpg files to pdf, I get a pdf file that has 3 images side by side on each page.
In addition, paper size settings are completely ignored.
If I use `convert -grayscale average`, then the pdf file is obtained as it should, except for the wrong sheet size. But `magick` command don't have ` -grayscale average` option. And Imagemagick 6 had no problems with grayscale jpg without ` -grayscale average`.

### Steps to Reproduce

1. Download attached image.
2. Try `convert *.jpg -quality 65 -compress jpeg -page A4 -density 72x72 stat.pdf` or `magick *.jpg -density 72  -page A4 -quality 65 -compress jpeg stat.pdf`.
3. Compare with `convert *.jpg -quality 65 -compress jpeg -page A4 -density 72x72 -grayscale average stat.pdf`.
4. Check paper size.

### Images

Source image:
![0](https://user-images.githubusercontent.com/20149245/150489669-bb301555-fa72-4d7b-8af2-e4e38aafd5c6.jpg)

Resulting pdf:
[stat.pdf](https://github.com/ImageMagick/ImageMagick/files/7911388/stat.pdf)

**Comment by fmw42** [comment]
Using your one JPG, your command works fine for me on IM 7.1.0-19 on Mac OSX Sierra

`magick 150489669-bb301555-fa72-4d7b-8af2-e4e38aafd5c6.jpg -quality 65 -compress jpeg -page A4 -density 72x72 stat.pdf`

[stat.pdf](https://github.com/ImageMagick/ImageMagick/files/7915051/stat.pdf)

**Comment by dymn** [comment]
Maybe, but the maximum version that is currently available in Gentoo is 7.1.0.14. The same bug.
IM build with this USE-flags: X bzip2 corefonts cxx djvu fontconfig jpeg lcms lzma openmp pango png postscript svg tiff truetype xml zlib -fftw -fpx -graphviz -hdri -heif -jbig -jpeg2k -lqr -opencl -openexr -perl -q8 -q32 -raw -static-libs -test -webp -wmf -zip

**Comment by snibgo** [comment]
In v7.1.0-4 and v7.1.0-20, in a directory that contains only the linked jpg, the command...
```
magick *.jpg -quality 65 -compress jpeg -page A4 -density 72x72 stat.pdf
```
... write a pdf with just one image, as expected.

@dymn wrote:
```
convert *.jpg ...
```
This reads _all_ the jpg files in the current directory, and writes them _all_ to a pdf. I suspect you have multiple jpg files in your current directory.

**Comment by dymn** [comment]
No, I know what I'm doing. If there are several jpg files in the folder, then a pdf is obtained in which the number of pages is equal to the number of jpg files, but each picture is displayed three times on the page, as in my example.
Could there be a problem with ghostscript-gpl-9.55.0-r1?

**Comment by fmw42** [comment]
ImageMagick does not use Ghostscript to create PDFs, only to read them.  Your issue is likely fixed in later releases of ImageMagick

**Comment by dymn** [comment]
The problem is solved in 7.1.0.20.
